docs/README.md

```diff
@@ -1,2 +1,294 @@
+- [ ] Check Status of TokenLocker and ERC721 and ERC1155 support (Bruce)
+- [ ] Review item which is needed for Permissionless Rollout of Validators (waiting on something on Ethereum?)
+- [ ] Check test updates from Boris
+
+## Environment Data
+
+Here is an overview of Environment data setup for testing after following the steps below
+
+* Deployer Address : 0x8875fc2A47E35ACD1784bB5f58f563dFE86A8451
+* Infura Project: 32cb9c57bfe447a99ea34e30195b2d10
+* KOVAN ERC20 Contract: [0xc90a6555CaD53a9D85a80052Fe2926E21608CF41](https://kovan.etherscan.io/address/0xc90a6555cad53a9d85a80052fe2926e21608cf41)
+* ROPSTEN ERC20 Faucet Contract
+
+# Developer and Testing guide
+## Setting up the codebase
+
+### Install the tools and the cli
+
+```
+cd ./tools
+cd elc
+yarn install
+cd ../eprover
+yarn install
+cd ../eth2hmy-relay
+yarn install
+cd ../../cli
+yarn install
+```
+
+*Note: you can see the CLI commands available by typing*
+`node index.js [command] -h`
+
+## Setting up the Infrastructure 
+
+### Infura Project Setup
+
+We use an an infura account to integrate with a Ropsten Node.
+
+Create an [Infura Account](https://infura.io/) and create an ethereum project. Add the INFURA_PROJECT_ID to the `.env` file.
+
+### Running a local Harmony Network
+See [here](https://github.com/harmony-one/harmony#debugging) for build instructions
+
+```
+cd $(go env GOPATH)/src/github.com/harmony-one/harmony
+make debug
+```
+To stop the network use `^C` or `make debug-kill`
+
+*Note: If using a later version of openssl (e.g. openssl v3.3) on a mac you may need to modify `scripts\go_executable_build.sh` changing this line `LIB[libcrypto.3.dylib]=/usr/local/opt/openssl/lib/libcrypto.3.dylib`*
+
+### Deployer Account
+Create a deployer account and fund it in both Harmony Testnet and Ropsten using Faucets. Add the PRIVATE_KEY to the `.env` file
+
+[Here](https://ropsten.oregonctf.org/) is a ropsten faucet.
+
+To fund your harmony account use the [harmony cli](https://docs.harmony.one/home/general/wallets/harmony-cli) or metamask and transfer funds from the following account.
+
+
+```
+On localnet, by default, 0xA5241513DA9F4463F1d4874b548dFBAC29D91f34 has funds, as defined in core/genesis.go. The private key for this address is 1f84c95ac16e6a50f08d44c7bde7aff8742212fda6e4321fde48bf83bef266dc
+```
+
+For testing purposes we used account `0x8875fc2A47E35ACD1784bB5f58f563dFE86A8451` and funded it with 1000 ONE on localnet and 1 ETH on Ropsten, but you can use any acccount you like as long as you know the private key or mnemonic.
+
+## Creating the DAG Merkle Tree
+
+DAG genertion takes several hours to run Ganesha has a machine to do this and shares the latest DAG info from Ropsten using [google drive](https://drive.google.com/file/d/1FqLCO5oc1xDYNMuub7xAqnb6kfohdf-U/view?usp=sharing). The epoch logic is the block Number divided by 30,000 so current Ropsten EPOCH is block 12280236 / 30000 = 409 which is the DAG info shared above.
```

Comment:
The code looks like something that can be parallelized and made efficient. What exactly is DAG and what is the purpose of generating all of them? Why is it aligned to epoch numbers? I think these questions should be answered before discussing how DAG is generated and the relevant commands